apps/src/blockly/addons/cdoFieldImageDropdown.js

```diff
@@ -109,6 +109,39 @@ export class CdoFieldImageDropdown extends FieldGridDropdown {
       this.dropdownDispose_.bind(this)
     );
   }
+
+  /**
+   * @override
+   * Ensure that the input value is a valid language-neutral option.
+   * The only change we made from the parent method is to not use the cache,
+   * as the list of options can change.
+   *
+   * @param newValue The input value.
+   * @returns A valid language-neutral option, or null if invalid.
+   */
+  doClassValidation_(newValue) {
```

Comment:
I confused myself by finding this: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/blockly/addons/cdoFieldDropdown.js

It looks like in this `doClassValidation_` function, we use: `const useCache = this.name !== 'BEHAVIOR';`

Is there anything from this similar customization we should consider here? Generally this seems reasonable to me!

Reply:
The name here wasn't very specific; for the background dropdown it was just `IMG`. So I decided it would make more sense to just do it for everything.

Reply:
I think Molly has the right instinct here. Standard dropdowns should be able to use the cache because the options aren't dynamic. The behavior picker is an exception and we can check for it based on the field name.

Image dropdowns on the other hand are always dynamic (at least assuming the costumes/backgrounds tabs are enabled). There are tons of costume-picker blocks and I would guess many have different field names. (In a quick look I found COSTUME, ANIMATON, TARGET, and GROUP all as potential field names.)

I can't imagine there are any major drawbacks to just using the cache for all of this type of field. In levels where they aren't dynamic (e.g. no costume tab, bounce) it should be a small-ish list of options any way.

Reply:
Thanks for the explanation, y'all!